File: CS-Examples/23_Worksheets/DetectEmptyWorksheet.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.IO;
using Spire.Xls;
using Spire.Xls.Charts;
using System.Text;
using System.Collections.Generic;

namespace DetectEmptyWorksheet
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ReadImages.xlsx");

            // Get the first worksheet from the workbook
            Worksheet worksheet1 = workbook.Worksheets[0];

            // Detect if the first worksheet is empty
            bool detect1 = worksheet1.IsEmpty;

            // Get the second worksheet from the workbook
            Worksheet worksheet2 = workbook.Worksheets[1];

            // Detect if the second worksheet is empty
            bool detect2 = worksheet2.IsEmpty;

            // Create a StringBuilder to save the content
            StringBuilder content = new StringBuilder();

            // Format the result string for displaying
            string result = string.Format("The first worksheet is empty or not: {0}\r\nThe second worksheet is empty or not: {1}", detect1, detect2);

            // Add the result string to the StringBuilder
            content.AppendLine(result);

            // Specify the output file path and name
            string outputFile = "Output.txt";

            // Save the result to a text file
            File.WriteAllText(outputFile, content.ToString());

            // Dispose of the workbook object to release resources
            workbook.Dispose();
  
            // Launch the file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: CS-Examples/23_Worksheets/DetectEmptyWorksheet.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts
Imports System.Text

Namespace DetectEmptyWorksheet
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file into the workbook from a specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ReadImages.xlsx")

            ' Get the reference to the first worksheet in the workbook
            Dim worksheet1 As Worksheet = workbook.Worksheets(0)

            ' Check if the first worksheet is empty or not and store the result in a Boolean variable
            Dim detect1 As Boolean = worksheet1.IsEmpty

            ' Get the reference to the second worksheet in the workbook
            Dim worksheet2 As Worksheet = workbook.Worksheets(1)

            ' Check if the second worksheet is empty or not and store the result in a Boolean variable
            Dim detect2 As Boolean = worksheet2.IsEmpty

            ' Create a StringBuilder object to hold the content
            Dim content As New StringBuilder()

            ' Create a formatted string with the results of the worksheet emptiness checks
            Dim result As String = String.Format("The first worksheet is empty or not: " & detect1 & vbCrLf & "The second worksheet is empty or not: " & detect2)

            ' Append the result to the content StringBuilder
            content.AppendLine(result)

            ' Specify the output file name for the text file
            Dim outputFile As String = "Output.txt"

            ' Write the content of the StringBuilder to the output file
            File.WriteAllText(outputFile, content.ToString())

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
